strawberry/types/type_resolver.py

```diff
@@ -376,7 +382,8 @@ def _get_fields(cls: Type) -> List[FieldDefinition]:
                 default_value=getattr(cls, field.name, undefined),
             )
 
-        field_name = cast(str, field_definition.origin_name)
-        field_definitions[field_name] = field_definition
+        if not isinstance(field_definition.type, Private):
```

Comment:
I would move this check inside of the `else` [here](https://github.com/strawberry-graphql/strawberry/pull/479/files#diff-23d017233da3ec1535492f9e4e36ef87R368-R383), and just `continue` if the `.type` is `Private` instead of going through the effort of creating the `FieldDefinition` and then abandoning it

Reply:
Very sad. The link doesn't appear to work. It's the `else` of the
```python
if hasattr(field, "_field_definition"): ...
else: ...
```
block right above

Reply:
Done